src/components/ebay-table/examples/sort-client-side.marko

```diff
@@ -0,0 +1,93 @@
+import data from "./data.json";
+import { type TableSort } from "../component";
+class {
+    declare state: {
+        sorted: Record<string, TableSort>;
+    };
+    onCreate() {
+        this.state = { sorted: {} };
+    }
+    onSort(event: { sorted: Record<string, TableSort> }, ...params: any) {
+        this.state.sorted = event.sorted;
+        this.emit("sort", event, ...params);
+    }
+}
+<ebay-table on-sort("onSort") ...input>
+    <@header name="sellerCol" column-type="row-header">
+        Seller
+    </@header>
+    <@header name="itemCol">
+        Item
+    </@header>
+    <@header name="statusCol">
+        Status
+    </@header>
+    <@header
+        name="listPriceCol"
+        column-type="numeric"
+        sort=(state.sorted.listPriceCol || "none")
+    >
+        List Price
+    </@header>
+    <@header
+        name="quantityCol"
+        column-type="numeric"
+        sort=(state.sorted.quantityCol || "none")
+    >
+        Quantity Available
+    </@header>
+    <@header name="orderCol">
+        Orders
+    </@header>
+    <@header name="watchersCol" column-type="numeric">
+        Watchers
+    </@header>
+    <@header name="protectionCol" column-type="numeric">
+        Protection
+    </@header>
+    <@header name="shippingCol">
+        Shipping
+    </@header>
+    <@header name="deliveryCol">
+        Delivery
+    </@header>
+    <for|r| of=data.sort((a, b) => {
```

Comment:
This will re-sort every rerender, I don't think it's okay to recommend this behavior even though technically in this example the only cause of a rerender is the sorting parameters changing. Maybe you can store this sorted data in `sorted` instead of the parameters for sorting?

Reply:
I'll use state